notebooks/timeseries.ipynb

```diff
@@ -0,0 +1,568 @@
+    "\n",
+    "In this notebook we go through the different interpolation and extrapolation options available to OpenSCM users as well as the different conventions OpenSCM understands. If your particular use case is not here, please raise an issue at [https://github.com/openclimatedata/openscm/issues]()."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# NBVAL_IGNORE_OUTPUT\n",
+    "from itertools import count\n",
+    "from copy import copy\n",
+    "\n",
+    "import numpy as np\n",
+    "import scipy.interpolate as interpolate\n",
+    "import scipy.integrate as integrate\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "from openscm import timeseries\n",
+    "from openscm.core import ParameterSet"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "COLOUR_1 = \"lime\"\n",
+    "COLOUR_2 = \"midnightblue\"\n",
+    "COLOUR_3 = \"firebrick\"\n",
+    "COLOUR_4 = \"deeppink\"\n",
+    "\n",
+    "MARKER_1 = \"o\"\n",
+    "MARKER_2 = \"^\"\n",
+    "MARKER_3 = \"+\"\n",
+    "MARKER_4 = \"x\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Holding data\n",
+    "\n",
+    "In OpenSCM, all data is stored in instances of `ParameterSet`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "paras = ParameterSet()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "## Example 1: Period average data with linear interpolation\n",
+    "\n",
+    "As it stands, the only case OpenSCM can handle is linear interpolation of data where the convention is that the source values represent the average of the underlying timeseries over a time period. This time period extends from the time time at which the value is provided, the 'data time', up until the next time in the timeseries. A common example for such a convention is emissions data, where each time point represents the average emissions flux for that time period e.g. 2100 emissions of 3 GtC / yr means that the average rate of emissions for 2100 was 3 GtC / yr.\n",
+    "\n",
+    "<div class=\"alert alert-warning\">\n",
+    "  <strong>Warning!</strong> Emissions data is ocassionally reported without the rate e.g. 2100 emissions were 3 GtC. In OpenSCM, we explicitly do not allow this as it doesn't actually make sense. The cumulative emissions for the year were 3 GtC but the emissions were not 3 GtC at any given point in 2100. Before passing emissions data into OpenSCM it should have the units converted to rates (of course, cumulative emissions data should not be reported as rates and so perhaps the only fix required is changing the variable name from `Emissions` to `Cumulative Emissions`).\n",
```

Comment:
```suggestion
    "  <strong>Warning!</strong> Emissions data is ocassionally reported without the rate e.g. 2100 emissions were 3 GtC. Before passing emissions data into OpenSCM it should have the units converted to rates (of course, cumulative emissions data should not be reported as rates and so perhaps the only fix required is changing the variable name from `Emissions` to `Cumulative Emissions`).\n",
```
...we just don't support it. We could, in principle, convert to rates, but setting a standard here is much easier.